bam_addrprg.c

```diff
@@ -354,6 +354,11 @@ static bool init(const parsed_opts_t* opts, state_t** state_out) {
             fprintf(stderr, "[init] Error adding RG line with ID:%s to the output header.\n", opts->rg_id);
             return false;
         }
+        if (opts->mode == overwrite_all &&
+            -1 == sam_hdr_keep_line(retval->output_header, "RG", "ID", opts->rg_id)) {
```

Comment:
Thanks for fixing this.

It's a bit surprising that a function called `sam_hdr_keep_line` has the effect of removing lines! Maybe this API function could be renamed to `sam_hdr_keep_only_line` or similar?

Reply:
Renamed to `sam_hdr_remove except`.